src/java.base/share/classes/jdk/internal/access/SharedSecrets.java

```diff
@@ -70,13 +70,27 @@
     private static JavaNetUriAccess javaNetUriAccess;
     private static JavaNetURLAccess javaNetURLAccess;
     private static JavaNioAccess javaNioAccess;
+    private static JavaUtilCollectionAccess javaUtilCollectionAccess;
     private static JavaUtilJarAccess javaUtilJarAccess;
     private static JavaUtilZipFileAccess javaUtilZipFileAccess;
     private static JavaUtilResourceBundleAccess javaUtilResourceBundleAccess;
     private static JavaSecurityAccess javaSecurityAccess;
     private static JavaSecuritySignatureAccess javaSecuritySignatureAccess;
     private static JavaxCryptoSealedObjectAccess javaxCryptoSealedObjectAccess;
 
+    public static void setJavaUtilCollectionAccess(JavaUtilCollectionAccess juca) {
+        javaUtilCollectionAccess = juca;
+    }
+
+    public static JavaUtilCollectionAccess getJavaUtilCollectionAccess() {
+        if (javaUtilCollectionAccess == null) {
+            try {
+                Class.forName("java.util.ImmutableCollections$Access", true, null);
```

Comment:
How does this work ? It attempts to load this class but `javaUtilCollectionAccess` is never assigned to a new value.

Reply:
Yeah the SharedSecrets stuff is pretty twisted